### PR TITLE
add missing cpl_vsi_error.h

### PR DESCRIPTION
**Pull request by ocefpaf**
@conda-forge/libgdal the absence of `cpl_vsi_error.h` is preventing us from building the `gdal` python bindings. Not sure if the absence of this file is by design or a mistake upstream.

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -92,6 +92,9 @@ if errorlevel 1 exit 1
 copy *.lib %LIBRARY_LIB%\ || exit 1
 if errorlevel 1 exit 1
 
+# libgdal 2.3.0
+copy %SRC_DIR%\port\cpl_vsi_error.h %LIBRARY_PREFIX%
```
Comment (gillins) [inline recipe/bld.bat]: Should this be `%LIBRARY_INC%` instead?

**Comment by gillins** [comment]
> Not sure if the absence of this file is by design or a mistake upstream.

Seems like a mistake upstream: https://github.com/OSGeo/gdal/issues/674

File: recipe/bld.bat (hunk 1 above)
Comment (ocefpaf) [reply recipe/bld.bat]: Sorry, copy and pasta (this is what I get for sending PRs late at night :smile:)

**Comment by ocefpaf** [comment]
@gillins we have a new error that is causing a segfault :-/

But I did notice that we got `OpenJPEG support: no` and that is probably the reason for the lack of this driver in the python bindings.

**Comment by kmuehlbauer** [comment]
@ocefpaf Maybe related: https://lists.osgeo.org/pipermail/gdal-dev/2018-June/048657.html

**Comment by ocefpaf** [comment]
@kmuehlbauer that did the trick, thanks! But I'm still puzzled by the segfault :-/

**Comment by ocefpaf** [comment]
Here are the erros:

```
libtool:   error: you must specify an output file
/bin/sh: line 3: 30257 Segmentation fault      (core dumped) /opt/conda/conda-bld/libgdal_1529500577956/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libhdf5.so
/bin/sh: line 3: 30258 Segmentation fault      (core dumped) /usr/lib64/librt.so
/bin/sh: /usr/lib64/libpthread.so: Permission denied
/bin/sh: line 3: 30260 Segmentation fault      (core dumped) /opt/conda/conda-bld/libgdal_1529500577956/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libz.so
/bin/sh: line 3: 30261 Segmentation fault      (core dumped) /usr/lib64/libdl.so
/bin/sh: line 3: 30262 Segmentation fault      (core dumped) /usr/lib64/libm.so -o libgdal.la ./ogr/gml2ogrgeometry.lo ./ogr/ogr2gmlgeometry.lo ./ogr/ogr_api.lo ./ogr/ogr_expat.lo ./ogr/ogr_fromepsg.lo ./ogr/ogr_geo_utils.lo ./ogr/ogr_geocoding.lo ./ogr/ogr_opt.lo ./ogr/ogr_srs_dict.lo ./ogr/ogr_srs_erm.lo ./ogr/ogr_srs_esri.lo ./ogr/ogr_srs_ozi.lo ./ogr/ogr_srs_panorama.lo ./ogr/ogr_srs_pci.lo ./ogr/ogr_srs_proj4.lo ./ogr/ogr_srs_usgs.lo ./ogr/ogr_srs_validate.lo ./ogr/ogr_srs_xml.lo ./ogr/ogr_srsnode.lo ./ogr/ogr_xerces.lo ./ogr/ograpispy.lo ./ogr/ograssemblepolygon.lo ./ogr/ogrcircularstring.lo ./ogr/ogrcompoundcurve.lo ./ogr/ogrct.lo ./ogr/ogrcurve.lo ./ogr/ogrcurvecollection.lo ./ogr/ogrcurvepolygon.lo ./ogr/ogrfeature.lo ./ogr/ogrfeaturedefn.lo ./ogr/
```

Not sure yet what is going on, this was passing yesterday though.

**Comment by kmuehlbauer** [comment]
@ocefpaf 

Can you explain, why some of the compile artifacts are linked with relative paths (`./ogr/gml2ogrgeometry.lo`) and others with full path (`/feedstock_root/build_artefacts/libgdal_1529500300408/work/gdal-2.3.0/alg/contour.lo`) in the linux build?

**Comment by gillins** [comment]
This seems related: https://bitbucket.org/chchrsc/kealib/issues/3/kea-config-hdflibs-wrong-format. Problem was introduced in most recent `kealib`. I'll see about patching `kealib` tomorrow to workaround it, but disabling it for Unix seems to fix the problem for now...

**Comment by ocefpaf** [comment]
Thanks @gillins!

**Pull request merged**